### PR TITLE
Add unit test of successful downloads

### PR DESCRIPTION
**Pull request by pontus**
Add unit test for successful downloads as well.

### CODE REVIEW

File: sda-download/api/sda/sda_test.go [hunk 1]
```diff
@@ -552,3 +566,217 @@ func TestEncrypted_Coords(t *testing.T) {
...
+	w = httptest.NewRecorder()
+	c, _ = gin.CreateTestContext(w)
+	c.Request = &http.Request{Method: "GET", URL: &url.URL{Path: "/s3-encrypted/somepath", RawQuery: "filename=somepath"}}
+	c.Request.Header = http.Header{"Client-Public-Key": []string{config.Config.App.Crypt4GHPublicKeyB64},
+		"Range": []string{"bytes=0-10"}}
+
+	c.Params = make(gin.Params, 1)
+	c.Params[0] = gin.Param{Key: "type", Value: "encrypted"}
+
+	// Test the outcomes of the handler
```
Comment (nanjiangshu) [inline sda-download/api/sda/sda_test.go:753]: Maybe this comment can be more informative?
So as the rest of comments `// Test the outcomes of the handler`

File: sda-download/api/sda/sda_test.go [hunk 2]
```diff
@@ -552,3 +566,217 @@ func TestEncrypted_Coords(t *testing.T) {
...
+
+	err = binary.Write(w, binary.BigEndian, int32(len(response)))
+	assert.NoError(f.t, err, "Could not write response length")
+
+	_, err = w.Write(response)
+	assert.NoError(f.t, err, "Could not write response")
+	w.Header().Add("Grpc-status", "0")
+}
+
+func TestDownload_Success_Whole(t *testing.T) {
```
Comment (nanjiangshu) [inline sda-download/api/sda/sda_test.go:614]: It seems this test function contains many scenarios. Not sure if it is better to divide them into separate functions, but if we keep them in one function, maybe change the function name to e.g. `Test_Download_Whole_Or_Range` to better reflect the testing scenarios?

**Review by nanjiangshu — COMMENTED** [review]
The tests worked well when I tested! I just left a few minor comments.

**Review by MalinAhlberg — DISMISSED** [review]
Looks good! I agree with @nanjiangshu's comments, but it's not stopping.

File: sda-download/api/sda/sda_test.go (hunk 1 above)
Comment (pontus) [reply sda-download/api/sda/sda_test.go]: Yes, I've updated/added comments a bit now, good?

File: sda-download/api/sda/sda_test.go (hunk 2 above)
Comment (pontus) [reply sda-download/api/sda/sda_test.go]: Yes, it turned out to be a bit more setup than I intended from the start so it felt like it made sense to test more cases. I guess one could factor most of that out but there'll still be a fair bit of setup so I think it's best to keep it as is for now, updated the name.

**Review by MalinAhlberg — APPROVED** [review]
:+1:

**Pull request merged**